docs/architecture/adr-065-store-v2.md

```diff
@@ -81,73 +82,71 @@ less intrusive. Specifically, we propose to:
 
-Furthermore, we will keep the IAVL is the backing [commitment](https://cryptography.fandom.com/wiki/Commitment_scheme)
-store for the time being. While we might not fully settle on the use of IAVL in
+Furthermore, we will keep IAVL as the default [SC](https://cryptography.fandom.com/wiki/Commitment_scheme)
+backend for the time being. While we might not fully settle on the use of IAVL in
 the long term, we do not have strong empirical evidence to suggest a better
 alternative. Given that the SDK provides interfaces for stores, it should be sufficient
 to change the backing commitment store in the future should evidence arise to
 warrant a better alternative. However there is promising work being done to IAVL
 that should result in significant performance improvement <sup>[1,2]</sup>.
 
+Note, we will provide applications with the ability to use IAVL v1 and IAVL v2 as
+either SC backend, with the latter showing extremely promising performance improvements
+over IAVL v0 and v1, at the cost of a state migration.
+
 ### Separating SS and SC
 
 By separating SS and SC, it will allow for us to optimize against primary use cases
 and access patterns to state. Specifically, The SS layer will be responsible for
-direct access to data in the form of (key, value) pairs, whereas the SC layer (IAVL)
+direct access to data in the form of (key, value) pairs, whereas the SC layer (e.g. IAVL)
 will be responsible for committing to data and providing Merkle proofs.
 
-Note, the underlying physical storage database will be the same between both the
-SS and SC layers. So to avoid collisions between (key, value) pairs, both layers
-will be namespaced.
-
 #### State Commitment (SC)
 
-Given that the existing solution today acts as both SS and SC, we can simply
-repurpose it to act solely as the SC layer without any significant changes to
-access patterns or behavior. In other words, the entire collection of existing
-IAVL-backed module `KVStore`s will act as the SC layer.
+A foremost design goal is that SC backends should be easily swappable, i.e. not
+necessarily IAVL.  To this end, the scope of SC has been reduced, it must only:
+
+* Provide a stateful root app hash for height h resulting from applying a batch
+  of key-value set/deletes to height h-1.
+* Fulfill (though not necessarily provide) historical proofs for all heights < h.
+* Provide an API for snapshot create/restore to fulfill state sync requests. 
+
+An SC implementation may choose not to provide historical proofs past height h - n (n can be 0)
+due to the time and space constraints, but since store v2 defines an API for historical
+proofs there should be at least one configuration of a given SC backend which
+supports this.
 
-However, in order for the SC layer to remain lightweight and not duplicate a
-majority of the data held in the SS layer, we encourage node operators to keep
-tight pruning strategies.
+Pruning (managing state bloat) is not considered a feature of store v2 or SC in
```

Comment:
what are the side effects of this? dont we do pruning async?

Reply:
I removed this paragraph altogether. I don't think it makes much sense. Pruning is a first class citizen in store v2.